### PR TITLE
Add a setup script

### LINKED ISSUE #86: Add setup.py script
We should add a `setup.py` script to allow our packages to be installed on any OS that has Python installed. It would also possibly help with automated testing.

https://docs.python.org/2.6/install/
https://docs.python.org/2.6/distutils/setupscript.html

Same issue for SSM: apel/SSM#31

### PR DESCRIPTION
**Pull request by gregcorbett**
Resolves #86 

Adds a setup.py script to APEL, allows APEL packages to be installed on any OS that has Python installed

### CODE REVIEW

**Comment by tofu-rocketry** [comment]
If an issue exists for a pull request, I prefer not to add a label to the pull request.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+"""A setup script for APEL.
+
+This script installs the APEL library, client, parsers and server. This
+should be similar to installing the RPMs for apel-lib, apel-client,
+apel-parsers, and apel-server, although there may be some differences.
```
Comment (tofu-rocketry) [inline setup.py]: Does this not install the .pyc/o like the SSM setup.py? If so, should use a similar comment to SSM's.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                   'bin/apelauth'],
+          data_files=[(conf_dir, conf_files),
+                      (data_dir, schema_files),
+                      (data_dir, accounting_files),
+                      (data_dir, message_files),
+                      (data_dir, update_scripts),
+                      (log_rotate_dir, log_rotate_files)],
+          # zip_safe allows setuptools to install the project
+          # as a zipfile, for maximum performance!
+          zip_safe=True)
```
Comment (tofu-rocketry) [inline setup.py]: Let's stick this to `False` like ssm's setup.py.

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+    accounting_files = ['scripts/slurm_acc.sh', 'scripts/htcondor_acc.sh']
+
+    message_files = ['scripts/msg_status.py']
+
+    # log_rotate_files will later be copied to log_rotate_dir
+    log_rotate_dir = '/etc/logrotate.d'
+    log_rotate_files = ['scripts/apel-client']
+
+    # For 'python setup.py install' to
```
Comment (tofu-rocketry) [inline setup.py]: Should we have an `extras_require` like for ssm?

Comment (tofu-rocketry) [inline setup.py]: This did appear in the version before the rebase - did it get lost?

Comment (gregcorbett) [inline setup.py]: probably, can you post a link to the pre-rebase version?

Comment (tofu-rocketry) [inline setup.py]: Not to that exactly as I only have it locally, but it should be the same as for SSM:

https://github.com/apel/ssm/blob/dev/setup.py#L61

File: setup.py (hunk 1 above)
Comment (gregcorbett) [inline setup.py]: it does not, comment added

File: setup.py (hunk 3 above)
Comment (gregcorbett) [inline setup.py]: added

File: setup.py (hunk 2 above)
Comment (gregcorbett) [inline setup.py]: done

**Comment by gregcorbett** [comment]
changes made

**Comment by gregcorbett** [comment]
the inability to use `python setup.py test` does seem to be due to the use of `..` in `test/test_schemas.py`. presumably the tests aren't being run from the `test` directory, but rather the repo root, where `setup.py` is.

running `python ../setup.py test` from the `test` directory does successfully run the test suite, if one re-enables the test functionality of `setup.py`.

**Comment by tofu-rocketry** [comment]
Looks good. Could you rebase? Then I'll review and merge.

**Pull request merged**